internal/wasm/cache.go

```diff
@@ -67,15 +69,67 @@ type localFileCache struct {
 	// option sets for configuring the cache.
 	CacheOptions
 
+	// permissionCheckCache is a cache for permission check for private OCI images.
+	permissionCheckCache *permissionCache
+
 	// logger
 	logger logging.Logger
 }
 
+// permissionCache is a cache for permission check for private OCI images.
+type permissionCache struct {
```

Comment:
learn from https://github.com/istio/istio/pull/55168, should we use `otter` here?

Reply:
I think `otter` is over-skilled for this very simple permission check result cache :-)